add more FAQ information about annotations, labels, and taints

elmiko opened the pull request:
This change adds a new question and augements the current question on
labels. This is being proposed to fill out the user facing information
about how to properly apply annotations, labels, and taints to Machine
and Node objects. Some minor reformatting and coalescing has been done
to help maintain a consistent style.

elmiko commented:
/kind documentation

elmiko commented:
@michaelgugino ptal

elmiko commented:
i realize some of this information is repeated from https://github.com/openshift/machine-api-operator/blob/master/FAQ.md#adding-labels-to-nodes-via-machines , i wonder if i should try to combine them or maybe it's fine to repeat?

the former question is more verbose in its description.

elmiko commented:
@michaelgugino i took your suggestion and combined my proposed question with the previous one about labels. curious to hear your thoughts.

JoelSpeed commented on FAQ.md:
```diff
@@ -87,29 +89,33 @@ This is set via the MCO’s config, not the Machine-api. The config a Machine re
+MachineSets can be used to help mark the Machines and Nodes that are created from them by using Annotations and Labels. To do this you will need to add this information in various places on your MachineSet depending on where you would like this information. Please note that this information will only be applied to new Machines and Nodes created from the MachineSet, they will not be retroactively  ...
 
-To add labels to a Node on a MachineSet, the place where you should add labels is **.spec.template.spec.metadata.labels.**
+Setting labels or annotations in `MachineSet.Spec.Template.Metadata` will cause them to be applied to every Machine object created from that MachineSet. Specifically these labels and annotations will end up in `Machine.Metadata`.
```
Up to this point, `Labels` and `Annotations` have been capitalised, should we be consistent? 

In other places, we haven't included the type in field path strings, I think they have also been lowercased to reflect the yaml or json representation, eg `.spec.template.metadata`, should we stick with that convention or do you think this is more readable?

elmiko replied:
good questions, i think capitalizing the proper nouns for specific resources is probably a nice readability upgrade. but i could either way.

as for the field path strings, i like including the type just to be explicit about what resource is being referenced. i think i borrowed the capitalization from code or something, i can certainly revert to lowercase if it's easier. i should probably do that just to make the document consistent. i will also look at the openshift docs to see how they approach this.

JoelSpeed commented on FAQ.md:
```diff
@@ -87,29 +89,33 @@ This is set via the MCO’s config, not the Machine-api. The config a Machine re
 
-To add labels to a Node on a Machine that's not part of a MachineSet (and only
-    upon first creating the Machine object): **.spec.metadata.labels.**
+Setting labels or annotations in the `MachineSet.Spec.Template.Spec.Metadata` will cause them to be applied to every Machine and Node object created from the MachineSet. In the case of Machine objects, these labels and annotations will be applied in `Machine.Spec.Metadata`. For Node objects, these values will be applied to `Node.Metadata`.
```
I wonder if `applied to every machine` implies they would be on the top level metadata rather than in the spec? What is the connotation of a label being applied to something?

elmiko replied:
i wasn't trying to imply any deeper meaning, just trying to reflect what the user needs to set for something to end up in the related resource. i'm happy to reword this, maybe just say "added" instead of "applied"?

JoelSpeed replied:
It's fine as is, don't worry about changing

enxebre commented on FAQ.md:
```diff
@@ -87,29 +89,33 @@ This is set via the MCO’s config, not the Machine-api. The config a Machine re
-### Which labels get added to a Node?
-When we talk about Node labels here, it can be somewhat confusing.
+### Which Annotations and Labels get added to Nodes?
+MachineSets can be used to help mark the Machines and Nodes that are created from them by using Annotations and Labels. To do this you will need to add this information in various places on your MachineSet depending on where you would like this information. Please note that this information will only be applied to new Machines and Nodes created from the MachineSet, they will not be retroactively  ...
```
As it's today machines reconcile towards node anytime during its lifecycle https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/nodelink/nodelink_controller.go#L229-L239

The case where we only reconcile on creation is from machineSet to machines just like as any other machineSet field. Changes only affect new created replicas.

elmiko replied:
maybe we should word this paragraph differently. i think the message we are trying to convey is that updating a machineset won't trigger a retroactive application of annotations and labels.

elmiko commented:
/retest

elmiko commented:
updated to address the issues on review. mainly reformatting the field examples to be more consistent and also the capitalization. i also added some extra emphasis to the notes about when labels and annotations are applied.

JoelSpeed commented:
/approve

elmiko commented:
/remove lifecycle-stale

elmiko commented:
/remove-lifecycle stale

elmiko commented:
/retest

Danil-Grigorev reviewed (approved):
/lgtm
